unicorn/include/cocaine/zookeeper/exception.hpp

```diff
@@ -0,0 +1,36 @@
+/*
+* 2015+ Copyright (c) Anton Matveenko <antmat@yandex-team.ru>
+* All rights reserved.
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*/
+#ifndef ZOOKEEPER_EXCEPTION_HPP
+#define ZOOKEEPER_EXCEPTION_HPP
+
+#include <exception>
+#include <string>
+
+namespace zookeeper {
+class exception :
+    public std::exception
+{
+public:
+    exception(std::string message_prefix, int zk_error_code);
```

Comment:
Why exactly the string prefix? This exception class should transform all ZK errors in some kind of well-described manner?
Please, put here an example what `what()` returns.